src/main/java/io/cdap/plugin/gcp/gcs/actions/GCSArgumentSetter.java

```diff
@@ -0,0 +1,244 @@
+
+  private static final class Argument {
+
+    private String name;
+    private String type;
+    private JsonElement value;
+
+    Argument() {
+      type = "string";
+    }
+
+    public String getName() {
+      return name;
+    }
+
+    public void setName(String name) {
+      this.name = name;
+    }
+
+    public String getValue() {
+      if (value == null) {
+        throw new IllegalArgumentException("Null Argument value for name '" + name + "'");
+      }
+      if (type.equalsIgnoreCase("schema")) {
+        return createSchema(value).toString();
+      } else if (type.equalsIgnoreCase("int")) {
+        return Integer.toString(value.getAsInt());
+      } else if (type.equalsIgnoreCase("float")) {
+        return Float.toString(value.getAsFloat());
+      } else if (type.equalsIgnoreCase("double")) {
+        return Double.toString(value.getAsDouble());
+      } else if (type.equalsIgnoreCase("short")) {
+        return Short.toString(value.getAsShort());
+      } else if (type.equalsIgnoreCase("string")) {
+        return value.getAsString();
+      } else if (type.equalsIgnoreCase("char")) {
+        return Character.toString(value.getAsCharacter());
+      } else if (type.equalsIgnoreCase("array")) {
+        List<String> values = new ArrayList<>();
+        for (JsonElement v : value.getAsJsonArray()) {
+          values.add(v.getAsString());
+        }
+        return Joiner.on("\n").join(values);
+      } else if (type.equalsIgnoreCase("map")) {
+        List<String> values = new ArrayList<>();
+        for (Map.Entry<String, JsonElement> entry : value.getAsJsonObject().entrySet()) {
+          values.add(String.format("%s=%s", entry.getKey(), entry.getValue().getAsString()));
+        }
+        return Joiner.on(";").join(values);
+      }
+
+      throw new IllegalArgumentException("Invalid argument value '" + value + "'");
+    }
+
+    private Schema createSchema(JsonElement array) {
+      List<Schema.Field> fields = new ArrayList<>();
+      for (JsonElement field : array.getAsJsonArray()) {
+        Schema.Type type = Schema.Type.STRING;
+
+        JsonObject object = field.getAsJsonObject();
+        String fieldType = object.get("type").getAsString().toLowerCase();
+
+        boolean isNullable = true;
+        if (object.get("nullable") != null) {
+          isNullable = object.get("nullable").getAsBoolean();
+        }
+
+        String name = object.get("name").getAsString();
+
+        if (fieldType.equals("double")) {
+          type = Schema.Type.DOUBLE;
+        } else if (fieldType.equals("float")) {
+          type = Schema.Type.FLOAT;
+        } else if (fieldType.equals("long")) {
+          type = Schema.Type.LONG;
+        } else if (fieldType.equals("int")) {
+          type = Schema.Type.INT;
+        } else if (fieldType.equals("short")) {
+          type = Schema.Type.INT;
+        } else if (fieldType.equals("string")) {
+          type = Schema.Type.STRING;
+        }
+
+        Schema nullable;
```

Comment:
Nit: let's rename this to something like `schema`, since it might not be nullable

Reply:
@rmstar Done